File: untitled3/src/main/java/org/example/entity/OverlayPlugin.java
package org.example.entity;

import org.example.component.DaggerResolverComponent;

import javax.inject.Inject;

public class OverlayPlugin {
    @Inject
    public Resolver resolver;

    public void onCreate(Context context) {
        DaggerResolverComponent.builder().context(context).build().inject(this);
        System.out.println("onCreate: context is " + context);
        System.out.println("onCreate: resolver is " + resolver);
    }
}

File: untitled3/src/main/java/org/example/entity/OverlayPlugin.kt
package org.example.entity

import org.example.component.DaggerResolverComponent
import javax.inject.Inject

class OverlayPlugin {
    @JvmField
    @Inject
    var resolver: Resolver? = null
    fun onCreate(context: Context) {
        DaggerResolverComponent.builder().context(context).build().inject(this)
        println("onCreate: context is $context")
        println("onCreate: resolver is $resolver")
    }
}